### ISSUE TITLE
Incompatible type with non-mandatory `bool` arg and kwargs

### ISSUE THREAD

**Issue by srittau** [python/mypy#7572]
(Originally reported as python/typeshed#3270.)

Please consider:

```python
from typing import Any

def dec(x: bool = ..., **kwargs: Any) -> None: 
    pass

dec(**{"": 123})
```

With mypy 0.720 (Python 3.7.3) and 0.730 (Python 3.7.4) and mypy master this reports (no flags):

```
foo.py:6: error: Argument 1 to "dec" has incompatible type "**Dict[str, int]"; expected "bool"
```

**Comment by ashb** [comment]
We found this in our Airflow test suite -- for us the error only started appearing with 0.730

**Comment by ilevkivskyi** [comment]
I think this is a duplicate of https://github.com/python/mypy/issues/5382, possible solution suggested in https://github.com/python/mypy/issues/5382#issuecomment-431701846 should fix this.

**Comment by ashb** [comment]
@ilevkivskyi The original problem I ran in to from https://github.com/python/typeshed/issues/3270 was on `mock.patch` -- so I'm not quite sure where I should be putting the TypedDict?

```python
import unittest
from unittest import mock

class TestCloudBuildHookWithPassedProjectId(unittest.TestCase):
    @mock.patch(
        'airflow.gcp.hooks.base.GoogleCloudBaseHook.project_id',
        new_callable=mock.PropertyMock,
        **{"return_value.classify_text.return_value": {}}
    )
    def test_error_operation(self, _, mock_project_id):
        pass
```

(I'm not sure if it's relevant or not but this error doesn't show up on 0.720, only on 0.730.)

**Comment by ilevkivskyi** [comment]
I meant solution for us, not for users. But if you want a workaround, then I think something like this should pass (didn't test however):
```python
import unittest
from unittest import mock
from typing import Any
from typing_extensions import TypedDict

DummyMock = TypedDict('DummyMock', {"return_value.classify_text.return_value": Any})
dm: DummyMock = {"return_value.classify_text.return_value": {}}

class TestCloudBuildHookWithPassedProjectId(unittest.TestCase):
    @mock.patch(
        'airflow.gcp.hooks.base.GoogleCloudBaseHook.project_id',
        new_callable=mock.PropertyMock,
        **dm
    )
    def test_error_operation(self, _, mock_project_id):
        pass
```
If you need this for many keys you can probably add all of them to the same `TypedDict` and make it `total=False`.
